library/ssl_tls.c

```diff
@@ -2687,21 +2710,76 @@ int mbedtls_ssl_resend( mbedtls_ssl_context *ssl )
             ssl_swap_epochs( ssl );
```

Comment:
If we happen to enter the loop twice for the finished message, this will resulting in erroneously swapping the transform twice. At the moment, though, I can't conceive this happen realistically, because the finished message is small enough to be delivered in a single datagram.